packages/calypso-e2e/src/lib/components/sidebar-component.ts

```diff
@@ -40,64 +33,51 @@ export class SidebarComponent {
-	 * Note, in the current Nav Unification paradigm, clicking on certain combinations of sidebar menu items will trigger
-	 * navigation away to an entirely new page (eg. wp-admin). Attempting to reuse the SidebarComponent object
-	 * under this condition will throw an exception from the Playwright engine.
+	 * @returns the ElementHandle for the sidebar
+	 */
+	async waitForSidebarInitialization(): Promise< ElementHandle > {
+		return await this.page.waitForSelector( selectors.sidebar );
+	}
+
+	/**
+	 * Navigates to given (sub)item of the sidebar navigation.
 	 *
 	 * @param {{[key: string]: string}} param0 Named object parameter.
 	 * @param {string} param0.item Plaintext representation of the top level heading.
 	 * @param {string} param0.subitem Plaintext representation of the child level heading.
 	 * @returns {Promise<void>} No return value.
 	 */
-	async gotoMenu( { item, subitem }: { item?: string; subitem?: string } ): Promise< void > {
-		let selector;
-		const viewportName = getViewportName();
-
-		// Especially on mobile devices, there can be a race condition in clicking on "My Sites" button to slide in the sidebar,
-		// and that sidebar actually being initialized! So we want to wait and make sure the sidebar is actually in the DOM before proceeding.
-		const sidebar = await this.waitForSidebarInitialization();
-
-		// If mobile, sidebar is hidden by default and focus is on the content.
-		// The sidebar must be first brought into view.
-		if ( viewportName === 'mobile' ) {
+	async gotoMenu( { item, subitem }: { item: string; subitem?: string } ): Promise< void > {
+		if ( getViewportName() === 'mobile' ) {
 			await this._openMobileSidebar();
 		}
 
-		if ( item ) {
-			item = toTitleCase( item ).trim();
-			// This will exclude entries where the `heading` term matches multiple times
-			// eg. `Settings` but they are sub-headings in reality, such as Jetpack > Settings.
-			// Since the sub-headings are always hidden unless heading is selected, this works to
-			// our advantage by specifying to match only visible text.
-			selector = `${ selectors.heading } span:has-text("${ item }"):visible`;
-			await this._click( selector );
-		}
+		const itemSelector = `.sidebar >> text="${ item }"`;
+		await Promise.all( [ this.page.waitForNavigation(), this.page.click( itemSelector ) ] );
 
 		if ( subitem ) {
-			subitem = toTitleCase( subitem ).trim();
-			// If there is a subheading, by definition the expanded menu element will always be present.
-			await sidebar.waitForSelector( selectors.expandedMenu );
-			// Explicitly select only the child headings and combine with the text matching engine.
-			// This works better than using CSS pseudo-classes like `:has-text` or `:text-matches` for text
-			// matching.
-			selector = `${ selectors.subheading } >> text="${ subitem }"`;
-			await this._click( selector );
+			await Promise.all( [
+				this.page.waitForNavigation(),
+				this.page.click( `${ itemSelector } >> text="${ subitem }"` ),
```

Comment:
I've noticed that you're using the `page.click` methods now - does this mean the helper method `_click` isn't used anymore?

Reply:
Good question. That method was important mostly for the custom scroll-to-item behavior, and I'm not sure if we still need it. I'll run some checks on that and verify.

Reply:
Yep, that one was still needed. Tweaked it a bit and added back in https://github.com/Automattic/wp-calypso/pull/54987/commits/6c7f9f45e493a1779e612c04bf052280d1b15d91